packages/workspace/src/core/hasher/git-hasher.ts

```diff
@@ -167,3 +171,17 @@ export function getFileHashes(path: string): Map<string, string> {
     return new Map<string, string>();
   }
 }
+
+/**
+ * This utility is used to return a Map of filenames to hashes, where those filenames come from
+ * git's knowledge of:
+ *
+ * - files which are untracked (newly created)
+ * - files which are modified in some way (but NOT deleted) and either staged or unstaged
+ */
+export function getUntrackedAndUncommittedFileHashes(
+  path: string
+): Map<string, string> {
+  const { status } = gitStatus(path);
```

Comment:
As discussed on slack, this simply leverages the existing utility as there was no performance improvement to be had in going with the `git diff` + `git ls-files` approach